src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -382,37 +340,35 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
     }
   }
 
-  // Filter definition for edge `type`, `startNode` & `endNode`
+  // Filter definition for node/edge `type`, `startNode` & `endNode`
   private def createEqualsAttributeFilter(
-      attribute: String,
-      struct: GenericRowWithSchema,
-      instanceType: InstanceType): Either[CdfSparkException, FilterDefinition] = {
-    val instanceTypeString = instanceType match {
-      case InstanceType.Edge => "edge"
-      case InstanceType.Node => "node"
-    }
+      attributeVector: Seq[String],
+      struct: GenericRowWithSchema): Either[CdfSparkException, FilterDefinition] =
     Try {
       val space = struct.getString(struct.fieldIndex("space"))
       val externalId = struct.getString(struct.fieldIndex("externalId"))
       FilterDefinition.Equals(
-        property = Vector(instanceTypeString, attribute),
+        property = attributeVector,
         value = FilterValueDefinition.Object(
           Json.obj("space" -> Json.fromString(space), "externalId" -> Json.fromString(externalId)))
       )
     }.toEither.leftMap { _ =>
       new CdfSparkIllegalArgumentException(
-        s"""Invalid filter value for: '$instanceTypeString $attribute'
+        s"""Invalid filter value for: '${attributeVector}'
            |Expecting a struct with 'space' & 'externalId' attributes, but found: ${struct.json}
            |""".stripMargin
       )
     }
-  }
 
-  // Filter definitions for "space" & "externalId" attributes for nodes & edges
+  // Filter definitions for attributes for nodes & edges
   private def createNodeOrEdgeCommonAttributeRef(
       instanceType: InstanceType,
       attribute: String): Seq[String] =
-    Vector(instanceType.productPrefix.toLowerCase(Locale.US), attribute)
+    if (attribute.equalsIgnoreCase("_type")) {
```

Comment:
nit: might also be an option to wrap only attribute value replacement in if-else and keep vector&instanceType in place